pkg/actions/nodegroup/upgrade.go

```diff
@@ -34,10 +34,10 @@ func (m *Manager) Upgrade(options managed.UpgradeOptions) error {
 		return managedService.UpgradeNodeGroup(options)
 	}
 
-	return m.upgradeAndWait(options)
+	return m.upgradeAndWait(options, wait)
```

Comment:
A cleaner way would be to split `upgradeAndWait` into two functions `upgrade` and `waitForUpgrade`, and call the latter one only if `wait` is set:

```suggestion
	if err := m.upgrade(options); err != nil {
		return err
	}
	if wait {
		return m.waitForUpgrade(options)
	}
	return nil
```

Reply:
Thanks @cPu1. Modified the code accordingly